components/support/sql/src/conn_ext.rs

```diff
@@ -218,6 +218,13 @@ pub trait ConnExt {
     fn unchecked_transaction(&self) -> SqlResult<UncheckedTransaction<'_>> {
         UncheckedTransaction::new(self.conn(), TransactionBehavior::Deferred)
     }
+
+    /// Begin `unchecked_transaction` with `TransactionBehavior::Immediate`. Use
+    /// when the first operation will be a read operation, that further writes
+    /// depend on for correctness.
+    fn unchecked_transaction_imm(&self) -> SqlResult<UncheckedTransaction<'_>> {
```

Comment:
nit: I'd prefer to see something slightly more verbose -- `unchecked_transaction_immediate`

+1 on the comment though about when it is appropriate to use this over `unchecked_transaction`

Reply:
Hmm, rust tends to abbreviate things often -- `fn`, `Vec`, `SeqCst`, `dyn`, etc. We don't always follow this, but `unchecked_transaction_immediate` is getting up there in length, and I don't think it makes things substantially clearer than `unchecked_transaction_imm`

Reply:
Hah... no worries... that's my swift perspective showing :-P